tpu/test/queuedResource.test.js

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const path = require('path');
+const assert = require('node:assert/strict');
+const {describe, it} = require('mocha');
+const cp = require('child_process');
+const {TpuClient} = require('@google-cloud/tpu').v2alpha1;
+
+const execSync = cmd => cp.execSync(cmd, {encoding: 'utf-8'});
+const cwd = path.join(__dirname, '..');
+
+async function waitForTPUCreation(nodeName, zone) {
+  const tpuClient = new TpuClient();
+  const projectId = await tpuClient.getProjectId();
+
+  // Give a time to start process of creating TPU Node
```

Comment:
whole process of creation and deletion can take even ~10min, that's why I put 1min timeouts between calls